File: frontend/src/types/specialDay.ts
export type SpecialDayType = 'hard_deadline' | 'milestone' | 'rest_day' | 'custom';

export interface SpecialDay {
  id: number;
  date: string;
  type: SpecialDayType;
  label: string;
  color: string;
  icon: string;
  reminder: boolean;
  note?: string | null;
}

export interface SpecialDayCreate {
  date: string;
  type: SpecialDayType;
  label?: string;
  color?: string;
  icon?: string;
  reminder?: boolean;
  note?: string | null;
}

File: backend/app/schemas/special_day.py
"""Pydantic schemas for SpecialDay CRUD."""

from datetime import date
from typing import Optional
from pydantic import BaseModel, Field


class SpecialDayCreate(BaseModel):
    date: date
    type: str = Field(..., pattern=r"^(hard_deadline|milestone|rest_day|custom)$")
    label: str = ""
    color: str = Field(default="#ef4444", max_length=7)
    icon: str = ""
    reminder: bool = True
    note: Optional[str] = None


class SpecialDayUpdate(BaseModel):
    type: Optional[str] = Field(None, pattern=r"^(hard_deadline|milestone|rest_day|custom)$")
    label: Optional[str] = None
    color: Optional[str] = Field(None, max_length=7)
    icon: Optional[str] = None
    reminder: Optional[bool] = None
    note: Optional[str] = None


class SpecialDayResponse(BaseModel):
    id: int
    date: date
    type: str
    label: str
    color: str
    icon: str
    reminder: bool
    note: Optional[str] = None

    model_config = {"from_attributes": True}
